### ISSUE TITLE
Release 7.0.400 breaks Github actions where `dotnet test` works against net462 on non-windows targets

### ISSUE THREAD

**Issue by fluffynuts** [dotnet/sdk#34605]
### Describe the bug
A recent update to the dotnet sdk binary blob used by GitHub Actions has caused my test runs to fail:

Last passing run: https://github.com/fluffynuts/NExpect/actions/runs/5763231844/job/15624618499
First failing run: https://github.com/fluffynuts/NExpect/actions/runs/5854022299/job/15869449387

### To Reproduce
I've been able to reproduce the failing problem:
```
Error:  FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'Microsoft.VisualStudio.TestPlatform.ObjectModel.EqtTrace' threw an exception. ---> System.TypeLoadException: Could not load type of field 'Microsoft.VisualStudio.TestPlatform.ObjectModel.PlatformEqtTrace:TraceSourceLevelsMap' (1) due to: Could not resolve type with token 01000019 from typeref (expected class 'System.Diagnostics.TraceLevel' in assembly 'System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089') assembly:System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 type:System.Diagnostics.TraceLevel member:(null)
```

by downloading
- https://dotnetcli.azureedge.net/dotnet/Sdk/7.0.306/dotnet-sdk-7.0.306-linux-x64.tar.gz
- https://dotnetcli.azureedge.net/dotnet/Sdk/7.0.400/dotnet-sdk-7.0.400-linux-x64.tar.gz

and using the unpacked `dotnet` in the root of each archive to run `dotnet test src/NExpect.sln` from within a checkout of https://github.com/fluffynuts/NExpect/

7.0.306 runs fine; 7.0.400 errors as above

### Exceptions (if any)
```
Testhost process for source(s) '/home/runner/work/NExpect/NExpect/src/NExpect.Matchers.AspNetMvc.Tests/bin/Debug/net462/NExpect.Matchers.AspNetMvc.Tests.dll' exited with error: Unhandled Exception:
System.TypeInitializationException: The type initializer for 'Microsoft.VisualStudio.TestPlatform.ObjectModel.EqtTrace' threw an exception. ---> System.TypeLoadException: Could not load type of field 'Microsoft.VisualStudio.TestPlatform.ObjectModel.PlatformEqtTrace:TraceSourceLevelsMap' (1) due to: Could not resolve type with token 01000019 from typeref (expected class 'System.Diagnostics.TraceLevel' in assembly 'System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089') assembly:System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 type:System.Diagnostics.TraceLevel member:(null)
   --- End of inner exception stack trace ---
  at Microsoft.VisualStudio.TestPlatform.TestHost.Program.Main (System.String[] args) [0x00033] in <af8fc7b1c76049579919cce3f8e01556>:0 
Error:  FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'Microsoft.VisualStudio.TestPlatform.ObjectModel.EqtTrace' threw an exception. ---> System.TypeLoadException: Could not load type of field 'Microsoft.VisualStudio.TestPlatform.ObjectModel.PlatformEqtTrace:TraceSourceLevelsMap' (1) due to: Could not resolve type with token 01000019 from typeref (expected class 'System.Diagnostics.TraceLevel' in assembly 'System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089') assembly:System, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 type:System.Diagnostics.TraceLevel member:(null)
   --- End of inner exception stack trace ---
  at Microsoft.VisualStudio.TestPlatform.TestHost.Program.Main (System.String[] args) [0x00033] in <af8fc7b1c76049579919cce3f8e01556>:0 
```
### Further technical details
dotnet --info
---
.NET SDK:
 Version:   7.0.400
 Commit:    73bf45718d
```
Runtime Environment:
 OS Name:     gentoo
 OS Version:  2.13
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /tmp/dotnet/7.0.400/sdk/7.0.400/

Host:
  Version:      7.0.10
  Architecture: x64
  Commit:       a6dbb800a4

.NET SDKs installed:
  7.0.400 [/tmp/dotnet/7.0.400/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.10 [/tmp/dotnet/7.0.400/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.10 [/tmp/dotnet/7.0.400/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_ROOT       [/opt/dotnet-sdk-bin-7.0]

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

environment:
---
Ubuntu 22.04.3 on GitHub actions. Though I could confirm the failure as well in a VM of Ubuntu 22.04.3 with dotnet-sdk-7.0 installed.

I'm not sure if this is the correct place to report this issue, so please redirect me if not.

**Comment by fluffynuts** [comment]
For now, I've pinned dotnet to 7.0.306 in my GitHub actions file, and have gone back to green test runs: https://github.com/fluffynuts/NExpect/actions/runs/5855602740

**Comment by fingers10** [comment]
Yeah this happens in my repo as well. i was using `7.0.x` in dotnet setup and my build starts to fail and now I have manually set it to `7.0.306` to make build successful.

**Comment by nohwnd** [comment]
This should be fixed in in the 7.0.401 https://github.com/microsoft/vstest/issues/4657
